packages/sign-client/src/controllers/engine.ts

```diff
@@ -400,8 +400,7 @@ export class Engine extends IEngine {
     const message = await this.client.core.crypto.encode(topic, payload);
     const opts = ENGINE_RPC_OPTS[method].req;
     this.client.history.set(topic, payload);
-    await this.client.core.relayer.publish(topic, message, opts);
-
+    this.client.core.relayer.publish(topic, message, opts);
```

Comment:
Sorry but only seeing this diff now. How come we don't await the `relayer.publish` step anymore here?

Reply:
`sendResult` and `sendError` still await `relayer.publish` so I'm not clear on whether this is accidental or intentionally different

Reply:
This is the change I discussed regarding the occasional race conditions where awaiting the publish was causing the `response` event to be fired before the listener for it is initialized causing the workflow to hang

Reply:
Ok thanks for context 👍 

Does it make sense to align this for `sendResult` and `sendError` in terms of behaviour or is this unique to `sendRequest`? I think if this is the exception we should probably comment it in the code, since it looks like a mistake at first sight/someone might "fix" it down the line.

Reply:
It's a unique case with `sendRequest` due to the above-mentioned issue happening only in this case. I think adding a comment would be adequate